libcodechecker/analyze/store_handler.py

```diff
@@ -228,25 +234,45 @@ def addCheckerRun(session, storage_session, command, name, tag, username,
 
             LOG.info('Removing previous analysis results ...')
             session.delete(run)
+            session.flush()
 
+            # The adding of the new run must happen on a separate session,
```

Comment:
It is not clear for me why do you need the separate session.

Reply:
The fact that the run is being stored into needs to be committed into the database. This has to happen separately from the actual storage operations, so that SQLAlchemy's session buffer and the transaction buffer is separate.

Server 1 begins to store `new run`. It `COMMIT`s the lock into the database. That is there. The normal storage session (which is wrapped over in the internal timeout handler in case the storage subprocess hangs) begins to transactionally store data.
If server 2 tries to store into `new run`, it will see that a lock is there, and thus prevent storing.
If server 1 is already storing and another request tries to store into `new run`, server 1 will locally see in its internal timeout manager that a store is already going, and prevent access.

Reply:
I have renamed the sessions to make it more apparent, not just `sess1` and `sess2`.